internal/config/config_test.go

```diff
@@ -0,0 +1,175 @@
+
+	t.Run("env var not set and spec empty", func(t *testing.T) {
+
+		assert.Equal(t, SetContainerImage("", CrunchyPostgresHA_PG10), "")
+	})
+
+	t.Run("image name from spec", func(t *testing.T) {
+
+		os.Setenv(CrunchyPostgresHA_PG10, "envVarImage")
+		assert.Equal(t, SetContainerImage("specImageName", CrunchyPostgresHA_PG10), "specImageName")
+	})
+
+	t.Run("get image from env var when image missing in spec", func(t *testing.T) {
+
+		os.Setenv(CrunchyPostgresHA_PG10, "envVarImage")
+		assert.Equal(t, SetContainerImage("", CrunchyPostgresHA_PG10), "envVarImage")
+	})
+
+	t.Run("env var and spec empty", func(t *testing.T) {
+
+		os.Setenv(CrunchyPostgresHA_PG10, "")
+		assert.Equal(t, SetContainerImage("", CrunchyPostgresHA_PG10), "")
+	})
+}
+
+func TestSetPGContainerImage(t *testing.T) {
+
+	// setup the test environment and ensure a clean teardown
+	tEnv, tClient, _ := setupTestEnv(t, "postgrescluster-controller")
+	ctx := context.Background()
+	// set namespace name
+	ns := &v1.Namespace{}
+	ns.GenerateName = "postgres-operator-test-"
+	assert.NilError(t, tClient.Create(ctx, ns))
+	t.Cleanup(func() {
+		assert.Check(t, tClient.Delete(ctx, ns))
+		teardownTestEnv(t, tEnv)
+	})
+	namespace := ns.Name
+
+	// set up cluster1
+	clusterName1 := "hippocluster1"
+
+	// set up test cluster1
+	cluster1 := &v1beta1.PostgresCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterName1,
+			Namespace: namespace,
+		},
+		Spec: v1beta1.PostgresClusterSpec{
+			Metadata:        &v1beta1.Metadata{},
+			PostgresVersion: 12,
+			InstanceSets:    []v1beta1.PostgresInstanceSetSpec{},
+		},
+	}
+
+	cluster1.SetGroupVersionKind(v1.SchemeGroupVersion.WithKind("postgrescluster"))
+	if err := tClient.Create(ctx, cluster1); err != nil {
+		t.Error(err)
+	}
+
+	t.Run("env var not set and spec empty", func(t *testing.T) {
+
+		assert.Equal(t, SetPGContainerImage(cluster1), "")
+	})
+
+	t.Run("env var and spec empty", func(t *testing.T) {
+
+		os.Setenv(CrunchyPostgresHA_PG12, "")
+		assert.Equal(t, SetPGContainerImage(cluster1), "")
+	})
+
+	t.Run("check env var when image missing in spec", func(t *testing.T) {
+
+		os.Setenv(CrunchyPostgresHA_PG12, "envVarImage")
+		assert.Equal(t, SetPGContainerImage(cluster1), "envVarImage")
+	})
+
+	t.Run("return image name from spec", func(t *testing.T) {
+
+		cluster1.Spec.Image = "specImageName"
+		os.Setenv(CrunchyPostgresHA_PG12, "envVarImage")
+		assert.Equal(t, SetPGContainerImage(cluster1), "specImageName")
+	})
+
+	t.Run("return GIS image name from env var", func(t *testing.T) {
+
+		cluster1.Spec.Image = ""
+		cluster1.Spec.Metadata.Annotations = map[string]string{
+			naming.PostGISVersion: "3.0",
+		}
+		os.Setenv(CrunchyPostgresHA_PG12_GIS_3_0, "envVarGISImage")
+		assert.Equal(t, SetPGContainerImage(cluster1), "envVarGISImage")
+	})
+
+	t.Run("return GIS image name from spec", func(t *testing.T) {
+
+		cluster1.Spec.Image = "specGISImageName"
+		os.Setenv(CrunchyPostgresHA_PG12_GIS_3_0, "envVarGISImage")
```

Comment:
⛏️ This test would be more clear if it set the PostGIS annotation as well. It's relying on the prior test as-is.

Reply:
Switched to using a spec value, but included the GIS version explicitly for clarity.